### PR TITLE
handle multiple styles in styles module (action-wise) and dynamically update buttons

### LINKED ISSUE #3705: Delete styles
Description of the problem.
It is related to the list of styles that appears on the light table, there are several buttons to interact with the list that exists, one of them is to delete, which only allows you to do it one by one and in the case of wanting to delete several styles at once, it is not possible.

That I would like to meet.

I would like to be able to choose several styles at once, be they 2 or 35 styles and be able to eliminate them at once, without having to do it tediously one at a time.

Additional context

It would not be necessary to add more buttons, it would only be necessary to be able to choose several styles in the container box by pressing the "shift" key, and in the case of having several styles selected, the option of applying the style to the photograph would be canceled.

Thanks

### PR DESCRIPTION
**Pull request by johnny-bit**
have you ever wanted to apply multiple styles at once?
or maybe delete multiple styles?
or export multiple stypes?

now you can!

fixes #3705 (and couple wishes)

### CODE REVIEW

File: src/libs/styles.c [hunk 1]
```diff
@@ -232,93 +269,142 @@ static void create_clicked(GtkWidget *w, gpointer user_data)
...
+    for (style = style_names; style != NULL; style = style->next)
     {
-      dt_styles_delete_by_name(name);
-      _gui_styles_update_view(d);
+      dt_styles_delete_by_name((char*)style->data);
     }
+    _gui_styles_update_view(d);
   }
+
+  g_list_free(style_names);
```
Comment (TurboGit) [inline src/libs/styles.c:357]: g_list_free_ful()

File: src/libs/styles.c [hunk 2]
```diff
@@ -232,93 +269,142 @@ static void create_clicked(GtkWidget *w, gpointer user_data)
...
-  dt_lib_styles_t *d = (dt_lib_styles_t *)user_data;
-  const char *name = get_style_name(d);
+  GtkTreeIter iter;
+  GtkTreeModel *model= gtk_tree_view_get_model(d->tree);
 
-  if(name)
+  GList *selected_styles = gtk_tree_selection_get_selected_rows(selection, &model);
+  GList *style_names = NULL;
+  GList *style = NULL;
+  for (style = selected_styles; style != NULL; style = style->next)
```
Comment (TurboGit) [inline src/libs/styles.c:315]: for(GList *style =

File: src/libs/styles.c [hunk 3]
```diff
@@ -219,6 +222,40 @@ static void _styles_row_activated_callback(GtkTreeView *view, GtkTreePath *path,
...
+
+  if(gtk_tree_selection_count_selected_rows(selection) == 0) return;
+
+  GtkTreeIter iter;
+  GtkTreeModel *model= gtk_tree_view_get_model(d->tree);
+
+  GList *selected_styles = gtk_tree_selection_get_selected_rows(selection, &model);
+  GList *style_names = NULL;
+  GList *style = NULL;
+  for (style = selected_styles; style != NULL; style = style->next)
```
Comment (TurboGit) [inline src/libs/styles.c:238]: for(GList *style = ...

File: src/libs/styles.c [hunk 4]
```diff
@@ -232,93 +269,142 @@ static void create_clicked(GtkWidget *w, gpointer user_data)
...
```
Comment (TurboGit) [inline src/libs/styles.c:374]: for(GList *style = ...

Comment (TurboGit) [reply src/libs/styles.c:374]: BTW, this very same loop is repeated in three places, probably better to put this code into a routine.

GList *_get_selected_styles(...).

You'll probably have to not   g_value_unset(&value) inside this routine and so use a g_list_free_full() for the styles_names list. See below.

File: src/libs/styles.c [hunk 5]
```diff
@@ -232,93 +269,142 @@ static void create_clicked(GtkWidget *w, gpointer user_data)
...
-      g_free(filedir);
+      dt_styles_save_to_file((char*)style->data, filedir, FALSE);
+      dt_control_log(_("style %s was successfully saved"), (char*)style->data);
     }
-    g_free(name);
-    gtk_widget_destroy(filechooser);
+    g_free(filedir);
   }
+  gtk_widget_destroy(filechooser);
+  g_list_free(style_names);
```
Comment (TurboGit) [inline src/libs/styles.c:407]: here: g_list_free_full()

**Comment by TurboGit** [comment]
I have scheduled this for 3.2, seems safe enough, sounds fine to you ?

**Comment by johnny-bit** [comment]
Yes! I'll do it soon! I'm fiddling with tagging module and same funzies for enabling/disabling buttons (it's the last one I promise! i'm not touching geo one!)

File: src/libs/styles.c (hunk 4 above)
Comment (johnny-bit) [reply src/libs/styles.c:374]: the g_value_unset is needed since the list isn't of GValue items but of gchar* items :)

**Comment by johnny-bit** [comment]
Done!

Additionally I've added some log messages so users know that apply went well ;) I found out I rather liked the fact that i can see apply, even if effects on thumbnails are negligible :)

**Pull request merged**